.gitignore

```diff
@@ -1,89 +1,158 @@
 *~
 
-# Totalview breakpoint files
-*.TVD.*breakpoints
-
-# Mac related
-*.dSYM
-.DS_Store
+# Default install location
+/mfem/
 
-# Generated files in the root MFEM directory
+# Generated files in main directory, config/ and docs/
 /deps.mk
-
-# Generated files in config/
 config/_config.hpp
 config/config.mk
-
-# Optional user configuration files in config/
-config/user.cmake
-config/user.mk
-
-# Generated documentation files in doc/
 doc/CodeDocumentation.conf
 doc/CodeDocumentation.html
 doc/CodeDocumentation
 
-# Generated executables - examples and miniapps
-examples/**/ex[1-9]
-examples/**/ex[1-9]p
-examples/**/ex[1-9][0-9]
-examples/**/ex[1-9][0-9]p
-miniapps/electromagnetics/joule
-miniapps/electromagnetics/tesla
-miniapps/electromagnetics/volta
-miniapps/meshing/klein-bottle
-miniapps/meshing/mesh-explorer
-miniapps/meshing/mobius-strip
-miniapps/meshing/shaper
-miniapps/performance/ex[1-9]
-miniapps/performance/ex[1-9]p
-miniapps/tools/display-basis
+# Totalview breakpoint files
+*.TVD.*breakpoints
+
+# OS-specific: Mac
+*.dSYM
+.DS_Store
+
+# Example and miniapp binaries and outputs
+
+examples/ex1
+examples/ex1p
+examples/ex2
+examples/ex2p
+examples/ex3
+examples/ex3p
+examples/ex4
+examples/ex4p
+examples/ex5
+examples/ex5p
+examples/ex6
+examples/ex6p
+examples/ex7
+examples/ex7p
+examples/ex8
+examples/ex8p
+examples/ex9
+examples/ex9p
+examples/ex10
+examples/ex10p
+examples/ex11p
+examples/ex12p
+examples/ex13p
+examples/ex14
+examples/ex14p
+examples/ex15
+examples/ex15p
+examples/ex16
+examples/ex16p
+examples/ex17
+examples/ex17p
 
-# Files generated by examples and miniapps
-# (this list is based on 'clean' rules in makefiles)
 examples/refined.mesh
 examples/displaced.mesh
-examples/**/mesh.*
+examples/mesh.*
 examples/ex5.mesh
-examples/**/Example5*
-examples/**/Example9*
+examples/Example5*
+examples/Example9*
 examples/Example15*
-examples/**/Example16*
+examples/Example16*
 examples/sphere_refined.*
-examples/**/sol.*
-examples/**/sol_u.*
-examples/**/sol_p.*
-examples/**/ex9.mesh
-examples/**/ex9-mesh.*
-examples/**/ex9-init.*
-examples/**/ex9-final.*
-examples/**/deformed.*
-examples/**/velocity.*
-examples/**/elastic_energy.*
+examples/sol.*
+examples/sol_u.*
+examples/sol_p.*
+examples/ex9.mesh
+examples/ex9-mesh.*
+examples/ex9-init.*
+examples/ex9-final.*
+examples/deformed.*
+examples/velocity.*
+examples/elastic_energy.*
 examples/mode_*
-examples/**/ex16.mesh
-examples/**/ex16-mesh.*
-examples/**/ex16-init.*
-examples/**/ex16-final.*
+examples/ex16.mesh
+examples/ex16-mesh.*
+examples/ex16-init.*
+examples/ex16-final.*
+
+examples/sundials/ex9
+examples/sundials/ex9p
+examples/sundials/ex10
+examples/sundials/ex10p
+examples/sundials/ex16
+examples/sundials/ex16p
+
+examples/sundials/ex9.mesh
+examples/sundials/ex9-mesh.*
+examples/sundials/ex9-init.*
+examples/sundials/ex9-final.*
+examples/sundials/Example9*
+examples/sundials/deformed.*
+examples/sundials/velocity.*
+examples/sundials/elastic_energy.*
```

Comment:
Need to add the files generated by `examples/sundials/{ex16,ex16p}` to both `make clean` and `.gitignore`.

Also: need to add `examples/{sundials,petsc}` to the top-level `make distclean`.

Reply:
I just checked and `make distclean` seems to perform a `make clean` inside `examples/{sundials/petsc}`. I think it's added here: https://github.com/mfem/mfem/blob/master/makefile#L347

Reply:
You are right. The first part still needs to be fixed though.